File: Data Structures & Algorithms/max-water-container/submission-1.js
class Solution {
    /**
     * @param {number[]} heights
     * @return {number}
     */
    maxArea(heights) {
        let i = 0;
        let j = heights.length - 1;
        let output = (j - i) * Math.min(heights[i], heights[j]);;
        while (i < j) {
            
            if (heights[i] < heights[j]) {
                i++;
            } else {
                j--;
            }
            output = Math.max(output, output = (j - i) * Math.min(heights[i], heights[j]));
        }
        return output;
    }
}

File: Data Structures & Algorithms/climbing-stairs/submission-1.java
class Solution {
    int[] output;
    public int climbStairs(int n) {
        output = new int[n + 1];
        return recursive(n);
    }

    private int recursive(int n) {
        if (output[n] != 0) {
            return output[n];
        } else if (n == 1) {
            output[n] = 1;
            return 1;
        } else if (n == 2) {
            output[n] = 2;
            return 2;
        }
        output[n - 1] = recursive(n - 1);
        output[n - 2] = recursive(n - 2);
        return output[n - 2] + output[n - 1];
    }
}
